consumer/consumererror/partialerror.go

```diff
@@ -66,3 +71,11 @@ func PartialMetricsError(err error, failedMetrics pdata.Metrics) error {
 func (err PartialError) GetMetrics() pdata.Metrics {
 	return err.failedMetrics
 }
+
+// IsPartial checks if an error was wrapped with a PartialError.
+func IsPartial(err error) bool {
+	if err != nil {
```

Comment:
Not your fault :), usually I prefer the code to test fo conditions and return, then apply the logic:

```
if err == nil {
  return false
}
return errors.As(err, &PartialError{})
```

Reply:
I prefer the same and considered swapping the order both here and in `IsPermanent` that I copied it from, but I figured it wouldn't change the amount or level of indenting in this case.  I can swap them though.

Reply:
I know it does not change, but still I read it easier:

* if nil (edge case) then this 
* function logic